src/Microsoft.VisualStudio.ProjectSystem.Managed/IO/Win32FileSystem.cs

```diff
@@ -46,24 +45,18 @@ public void CopyFile(string source, string destination, bool overwrite)
             File.Copy(source, destination, overwrite);
         }
 
-        public string ReadAllText(string path)
+        public async Task<string> ReadAllTextAsync(string path)
```

Comment:
Unneeded state machine too.

Reply:
This state machine is needed in order to ensure the disposal happens after the read completes. Ditto for `WriteAllTextAsync`.